gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/DefaultKeystoreService.java

```diff
@@ -357,6 +359,30 @@ public void removeCredentialForCluster(String clusterName, String alias) throws
     }
   }
 
+  @Override
+  public void removeCredentialsForCluster(String clusterName, Set<String> aliases) throws KeystoreServiceException {
+    synchronized (this) {
+      KeyStore ks = getCredentialStoreForCluster(clusterName);
+      if (ks != null) {
+        try {
+          // Delete all the entries
+          for (String alias : aliases) {
+            if (ks.containsAlias(alias)) {
+              ks.deleteEntry(alias);
+            }
+            removeFromCache(clusterName, alias);
```

Comment:
Instead of invalidating one entry at a time we can bulk invalidate them using [invalidateAll(keys)](https://www.javadoc.io/doc/com.github.ben-manes.caffeine/caffeine/1.2.0/com/github/benmanes/caffeine/cache/Cache.html#invalidateAll-java.lang.Iterable-) API

Reply:
Is there any benefit in doing it this way?

Reply:
Done.